### ISSUE TITLE
InvalidCastException when accessing enumerator

### ISSUE THREAD

**Issue by GW-FUB** [mbdavid/LiteDB#1223]
I'm trying to get a value from my collection. I made a totally new database and inserted a model  consisting only from plain getters and setters with default C# types (string, DateTime, int).

When I try to access this with some enumerator (I hunt it down from using FindAll().ToList()) I get the following error:

> System.InvalidCastException: Specified cast is not valid.
> 
> at System.Nullable1[T].Unbox (System.Object o) [0x0000d] in <58604b4522f748968296166e317b04b4>:0
> 
> at (wrapper dynamic-method) System.Object.lambda_method(System.Runtime.CompilerServices.Closure,object,object)
> 
> at LiteDB.BsonMapper.DeserializeObject (System.Type type, System.Object obj, LiteDB.BsonDocument value) [0x0008b] in <9e5fde0ddc6f45bfbe6bed2535394f2f>:0
> 
> at LiteDB.BsonMapper.Deserialize (System.Type type, LiteDB.BsonValue value) [0x00202] in <9e5fde0ddc6f45bfbe6bed2535394f2f>:0 
> 
> at LiteDB.BsonMapper.ToObject (System.Type type, LiteDB.BsonDocument doc) [0x00028] in <9e5fde0ddc6f45bfbe6bed2535394f2f>:0 
> 
> at LiteDB.BsonMapper.ToObject[T] (LiteDB.BsonDocument doc) [0x00000] in <9e5fde0ddc6f45bfbe6bed2535394f2f>:0 
> 
> at LiteDB.LiteCollection1+<Find>d__17[T].MoveNext () [0x00090] in <9e5fde0ddc6f45bfbe6bed2535394f2f>:0 
> 
> at DieselApp.Controller.Data.OrdersController.LoadOrdersAsync () [0x00257] in C:\PROJECTS\MRD\DieselAppV2\Frontend\DieselApp\DieselApp\Controller\Data\OrdersController.cs:45`

**Comment by GW-FUB** [comment]
I have to add that I use "Nullable<>" for every datatype (that is not nullable by default).

**Comment by GW-FUB** [comment]
I figured it out with the sourcecode.

I changed the datatype of an property from an enum to a plain int. The enum-string-value was still stored in the database and when the engine tried to parse the string to an int? it failed, of course.

The location is `BsonMapper.Deserialize.cs:DeserializeObject(Type, object, BsonDocument):Line 258`.
The function is `member.Setter(obj, this.Deserialize(member.DataType, val));`
